### PR TITLE
nc-news-6 

### PR DESCRIPTION
**Pull request by RonaldB123**

### CODE REVIEW

File: __tests__/integration-tests/app.test.js [hunk 1]
```diff
@@ -85,4 +85,51 @@ describe("nc-news", ()=>{
...
+
+    describe("GET /api/articles/:article_id/comments", ()=>{
+        test("200: Responds with status code", ()=>{
+            return request(app).get("/api/articles/1/comments").expect(200);
+        })
+        test("200: Responds with array of comments when given article_id", ()=>{
+            return request(app).get("/api/articles/1/comments").then(({body}) =>{
+                const {comments} = body;
+             
+                expect(comments).toBeSortedBy("created_at", {descending: false});
```
Comment (B7Bprog) [inline __tests__/integration-tests/app.test.js]: Comments should be served with the most recent first.

File: express/model/commentByArId-model.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const db = require("../../db/connection");
+
+exports.getCommentsData = (id) =>{
+    return db.query(`SELECT * FROM comments 
+    WHERE article_id = $1
+    ORDER BY created_at ASC`, [id]).then(({rows})=>{
```
Comment (B7Bprog) [inline express/model/commentByArId-model.js]: Most recent comments first please. (DESC)

File: __tests__/integration-tests/app.test.js [hunk 3]
```diff
@@ -85,4 +85,51 @@ describe("nc-news", ()=>{
...
+                comments.forEach(comment =>{
+                    expect(comment).toHaveProperty("comment_id", expect.any(Number));
+                    expect(comment).toHaveProperty("votes", expect.any(Number));
+                    expect(comment).toHaveProperty("created_at", expect.any(String));
+                    expect(comment).toHaveProperty("author", expect.any(String));
+                    expect(comment).toHaveProperty("body", expect.any(String));
+                    expect(comment).toHaveProperty("article_id", expect.any(Number));
+                })
+            })
+        })
```
Comment (B7Bprog) [inline __tests__/integration-tests/app.test.js:114]: Please add another test (200) where an article doesn't have any comments and it responds with an empty array.

File: express/controller/commentByArId-controller.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const {getCommentsData} = require("../model/commentByArId-model");
+
+exports.getCommentsByArticleId = (req, res, next) =>{
+    const {article_id} = req.params;
```
Comment (B7Bprog) [inline express/controller/commentByArId-controller.js:5]: You will need a way to check the reason for the query returning empty rows:

- article exists but has no comments.
- article does not exist.

For this think about using Promise.all()

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Ronald, please check my comments, refactor your code, then send us another PR with it. 👍
